### PR TITLE
Removed 'external_services.istio.component_status.components' duplicate labels.

### PR DESCRIPTION
**Pull request by hhovsepy**
### Describe the change

`external_services.istio.component_status.components` was expecting to have values for `istiod` and `ingress/egressgateway`, which was causing duplication for Istio component statuses, also duplication of labels in configuration and in code.
Now `external_services.istio.component_status.components` by default made empty, and `istiod` and `ingress/egressgateway` component statuses are read in code.
Functionality of  `external_services.istio.component_status.components` is left as deprecated for now, until will be removed, or will left as empty for possible custom components such as `eastwestgateway`.

### Steps to test the PR
Make sure you have 'istiod' without pods sunning.
Before:
Screenshot from 2024-07-15 14-08-00

After:
Screenshot from 2024-07-15 13-34-34

### Automation testing
Tests modified.

### Issue reference
https://github.com/kiali/kiali/issues/7501
https://github.com/kiali/kiali/issues/7524
Operator PR https://github.com/kiali/kiali-operator/pull/795

### CODE REVIEW

**Review by jshaughn — COMMENTED** [review]
@hhovsepy I see a difference in status, but for me I still see a duplicate, is that expected?  I didn't start with a clean CR, would that make a difference?

Before:
image
After:
image

**Comment by hhovsepy** [comment]
Hi @jshaughn , is there a 'external_services.istio.component_status.components' remaining in your local configuration file?

**Comment by jshaughn** [comment]
@hhovsepy I think I must have an issue with my test env.  Let me try again...

File: config/config.go [hunk 1]
```diff
@@ -688,23 +689,6 @@ func NewConfig() (c *Config) {
...
-							AppLabel: "istio-ingressgateway",
-							IsCore:   true,
-							IsProxy:  true,
-						},
-						{
-							AppLabel: "istiod",
-							IsCore:   true,
-							IsProxy:  false,
-						},
-					},
```
Comment (jshaughn) [inline config/config.go:707]: > Functionality of external_services.istio.component_status.components is left as deprecated for now, until will be removed, or will left as empty for possible custom components such as eastwestgateway.

Do we really want to leave this unused Config?  If we think it's going to be useful in the near future, then maybe, but otherwise I'm not sure it's worth keeping.  If nothing else I think a comment may be appropriate in the type definition.

File: mesh/api/testdata/test_mesh_graph.expected [hunk 2]
```diff
@@ -230,33 +230,12 @@
...
-                },
-                {
-                  "AppLabel": "istiod",
-                  "IsCore": true,
-                  "IsProxy": false,
-                  "IsMultiCluster": false,
-                  "Namespace": ""
-                }
-              ]
+              "Components": null
```
Comment (jshaughn) [inline mesh/api/testdata/test_mesh_graph.expected:233]: This doesn't look right.  I would have expected it to be omitted, or be an empty array.

**Review by jshaughn — COMMENTED** [review]
OK, sorry, on re-test it looks as expected: 
image

A couple of comments...

File: config/config.go (hunk 1 above)
Comment (hhovsepy) [reply config/config.go:707]: The reason it is left is, that there can be components such as "istio-eastwestgateway", or any other custom component users might want to be checked. So far we can mark it as deprecated.

File: mesh/api/testdata/test_mesh_graph.expected (hunk 2 above)
Comment (hhovsepy) [reply mesh/api/testdata/test_mesh_graph.expected]: yes, thanks, should be an empty array

**Review by jshaughn — APPROVED** [review]
Looks good!
